### DISCUSSION TITLE
Mongo 5.x and upwards

### DISCUSSION THREAD

**Discussion by Mike-devs** [pocoproject/poco#3875] [Q&A]
Hello, I tried to use a simple query in Mongo with Poco::MongoDB feature. It crashed with an exception saying it didn't find the field I was trying to get with the following code:

`		Poco::MongoDB::ResponseMessage& response = cursor.next(connection);
		for (;;)
		{
			for (Poco::MongoDB::Document::Vector::const_iterator it = response.documents().begin(); it != response.documents().end(); ++it)
			{
				std::cout << (*it)->get<std::string>("appid") << std::endl;
			}
`

In fact, the exception message is misleading. It's only when browsing the iterator in the debugger that deep down inside, I found that the Mongo driver is using pre 5.x protocol, which is completely different now in 5.x and 6.x
As these versions are now kind of "old" already, I wouldn't want to work on a Mongo 4.x (where the code works perfectly fine) for my project.

Questions:
- is it planned to update quickly this driver?
- is it planned to work on an async driver, since this is one is a blocking one?

**Comment by matejk** [comment]
The MongoDB driver is planed to be updated for release 1.13 (see #3484).
